osv/ecosystems/cran.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""CRAN helpers."""
+
+import requests
+from .. import semver_index
+
+from . import config
+from .helper_base import Ecosystem, EnumerateError
+
+class CRAN(Ecosystem):
+  """CRAN ecosystem helpers."""
+
+  # Use the Posit Public Package Manager API to pull both the current and archived versions for a specific package
+  # since CRAN doesn't natively support this functionality.
+  _API_PACKAGE_URL_POSIT_CRAN = 'https://packagemanager.posit.co/__api__/repos/2/packages/{package}'
+
+  def sort_key(self, version):
```

Comment:
Could you not then subclass `SemverEcosystem` instead of `Ecosystem` making this not necessary?

Reply:
Ah I should've seen this, thank you for the review, I'll switch it over.

Reply:
`SemverEcosystem` doesn't actually get the real package versions, `next_version` just returns the next semver version numerically.

Since Bioconductor doesn't truly use semver (it doesn't allow for prerelease `x.y.z-pre` versions) I think leaving it as is is best.